payu/models/access.py

```diff
@@ -300,6 +284,33 @@ def archive(self):
                                              model.start_date_nml_name)
                 f90nml.write(end_date_dict, end_date_path, force=True)
 
+            if model.model_type == 'cice':
+
+                # Copy supplemental restart files to RESTART path
+                for f_name in model.access_restarts:
+                    f_src = os.path.join(model.work_path, f_name)
+                    f_dst = os.path.join(model.restart_path, f_name)
+
+                    if os.path.exists(f_src):
+                        shutil.move(f_src, f_dst)
+
+                # Copy "cice_in.nml" from work path to restart.
+                work_ice_nml_path = os.path.join(
+                                        model.work_path,
+                                        model.ice_nml_fname
+                )
+                restart_ice_nml_path = os.path.join(
+                                        model.restart_path,
+                                        model.ice_nml_fname
+                )
+
+                if os.path.exists(work_ice_nml_path):
+                    shutil.copy2(work_ice_nml_path, restart_ice_nml_path)
+
+                # Check that run produced iced.YYYYMMDD restart file matching
+                # run end date.
+                model.find_matching_iced(model.restart_path, run_end_date)
```

Comment:
Can this be run earlier ? By this point some files have already been archived, I think you can run it using the mode.work_path before the restart files are moved

Reply:
Ah this can only be run once the correct `run_end_date` has been calculated, and so I'm not sure if it can be run earlier

Reply:
I could move it up a little though, just before
```python
# Copy supplemental restart files to RESTART path
                for f_name in model.access_restarts:
                    f_src = os.path.join(model.work_path, f_name)
                    f_dst = os.path.join(model.restart_path, f_name)
```

Reply:
What is happening is that half the restarts get moved the the archive folder, and `payu sweep` won't clean them up. They get found when trying to run again

Reply:
Hmm yeah that's not ideal. The top level `access.py` driver's `archive()` is always called after the submodels: https://github.com/payu-org/payu/blob/da047828f8eee8bd7746f0b7cb3df75044cb94b8/payu/experiment.py#L798-L803

and so things will always be partially archived as long as we run the check in the `access` driver, which has access to the correct end date.

Perhaps it's better to just omit this additional test in that case? Let me know what you think

Reply:
Could you force the cice submodel part to run first, before any other models archive ?

Reply:
We've decided to defer this part for now